ofrak_components/ofrak_components/uf2.py

```diff
@@ -0,0 +1,74 @@
+import logging
+from uf2 import uf2conv
+from dataclasses import dataclass
+
+from ofrak.resource import Resource
+from ofrak.component.unpacker import Unpacker
+from ofrak.component.packer import Packer
+from ofrak.component.identifier import Identifier
+from ofrak.core.binary import GenericBinary
+from ofrak_type.range import Range
+
+LOGGER = logging.getLogger(__name__)
+
+
+@dataclass
+class Uf2Data(GenericBinary):
+    """
+    A UF2 binary blob
+    """
+
+
+class Uf2Unpacker(Unpacker[None]):
+    """
+    UF2 unpacker.
+
+    Extracts the data from a UF2 packed file.
+    """
+
+    targets = (Uf2Data,)
+    children = (GenericBinary,)
+
+    async def unpack(self, resource: Resource, config=None):
+        """
+        Unpack a UF2 file.
+
+        UF2 files contain blocks of binary data.
+        """
+
+        # TODO: technically there's multiple types of data. At least
+        # investigate what they mean - sj
+
+        resource_data = await resource.get_data()
+        uf2_data = uf2conv.convert_from_uf2(resource_data)
```

Comment:
I'm not sure this approach works.

We'll want to extract the `familyid` and `appstartaddr` either as an analyzer or part of unpack step (https://github.com/microsoft/uf2/blob/162acf4e4cb7c8e9ec589c842d277dcb5c43db89/utils/uf2conv.py#L19).

The `familyid`, at a minimum, needs to be saved so that the packer uses it, no?

Reply:
I got ahead of myself with the PR, forgot to tag as TODO, so my checks didn't complain. I'll read up on it and either guard or test for it.

Reply:
Covered by new analyzer